src/mlpack/methods/edge_boxes/feature_extraction.hpp

```diff
@@ -0,0 +1,262 @@
+   */
+  void ConvTriangle(CubeType &InImage, const size_t radius);
+
+  /**
+   * finds maximum of numbers on cube axis and stores maximum values
+   * in MaxVal, locations of maximum values in Location
+   * @param mag Input Cube for which we want to find max values and location
+   * @param Location Stores the slice number at which max value occurs
+   * @param MaxVal Stores the maximum value among all slices for a given (row, col).
+   */
+  void MaxAndLoc(CubeType &mag, arma::umat &Location, MatType& MaxVal) const;
+
+  /**
+   * Computes Magnitude & Orientation of the Edges.
+   * Gradient of a function is a vector of partial derivatives in each direction.
+   * In this function the edges are calculated by applying the sobel filter on Image
+   * which is the same as finding the vectors of partial derivates.
+   * These "vectors" have a magnitude and a direction (orientation), which we 
+   * calculate in this function.
+   * @param InImage Input Image for which we calculate Magnitude & Orientation.
+   * @param Magnitude Magnitude of the Edges
+   * @param Orientation Orientation of the Edges
+   */
+  void Gradient(const CubeType& InImage, 
+         MatType& Magnitude,
+         MatType& Orientation);
+
+  /**
+   * Compute Histogram of Oriented Gradients ( count occurrences of gradient
+   * orientation in localized portions of an image. )
+   * @param Magnitude gradient magnitude
+   * @param Orientation gradient orientation
+   * @param downscale spatially downscaling factor
+   * @param interp: 1 (true) for interpolation over orientations
+   */
+  void Histogram(const MatType& Magnitude,
+                 const MatType& Orientation, 
+                 const size_t downscale,
+                 const size_t interp,
+                 CubeType& HistArr);
+ 
+  /**
+   * store the features as a window view for a 16 x 16 patch, 
+   * centered at each location in loc.
+   * @param channels Input candidate features
+   * @param loc Input locations
+   * @param features Output features at each 16 x 16 window centered at loc
+   */
+  void ViewAsWindows(const CubeType& channels, const arma::umat& loc,
+                     CubeType& features);
+
+  /**
+   * Regular Features are features directly indexing to the image locations.
+   * Each channel is of same size as that of image and captures a different
+   * faucet of information, like color, gradient and oriented gradient
+   * information at a patch x. Refer to the below paper for details:
+   * Sketch Tokens: A Learned Mid-level Representation for Contour 
+   * and Object Detection.
+   * Published in: Computer Vision and Pattern Recognition (CVPR),
+   * 2013 IEEE Conference 
+   * Author(s) : Joseph J. Lim ; C. Lawrence Zitnick ; Piotr Dollár
+   * @param channels input candidate regular features
+   * @param loc input locations
+   * @param RegFtr output Regular Features at that location.
+   */
+  void GetRegFtr(const CubeType& channels, const arma::umat& loc,
```

Comment:
Can you use the doxygen style to cite the paper. Here is an example: https://github.com/mlpack/mlpack/blob/master/src/mlpack/methods/ann/layer/dropout_layer.hpp